gemrb/core/Map.cpp

```diff
@@ -2403,11 +2403,9 @@ void Map::PlayAreaSong(int SongType, bool restart, bool hard) const
 	// This assumption is definitely wrong for IWD, see #1476! Therefore we
 	// use a preliminary flag test to restrict it to BG1 for now.
 	if (IsStar(*poi) && !MasterArea && SongType == SONG_BATTLE && core->HasFeature(GFFlags::BREAKABLE_WEAPONS)) {
-		static constexpr int bc1Idx = 19; // fallback to first BG1 battle music
-
 		const Map* lastMasterArea = game->GetMap(game->LastMasterArea, false);
 
-		pl = lastMasterArea ? lastMasterArea->SongList[SongType] : bc1Idx;
```

Comment:
I don't understand the motivation here. There's a minor chance this gets hit and still we intentionally make it worse?

Reply:
I missed the forest for the trees here... My original motivation was a situation in the temple on Friendly Arm map (AR2304), where I started a battle with Gellana and no battle music is starting. The reason is a yet not fully understood behaviour of the original games, because AR2304 is itself a masterarea with no own battle music but should play the battle music of the "parent" masterarea AR2300. This situation is currently not covered here. Somehow for this commit I thought it was something different and AR2304 isn't a masterarea and the fallback occluded this from me. But as it turned out this is not the case here...

Still, with this change it becomes more obvious if something went wrong and further investigation is needed from the dev team. A fallback might occlude this and keep a deeper problem undiscovered.

Reply:
Either way you'd have had to step through the function, so I'd leave it in.

Checking PreviousArea instead should fix that case for most maps in bg1. The counterpoint to test would be Nashkel mines and similar, where you stay in the same worldmap location through several areas that are all likely in mastarea.2da.

Reply:
> Either way you'd have had to step through the function, so I'd leave it in.

Ok. I'll drop this commit then.

> Checking PreviousArea instead should fix that case for most maps in bg1. The counterpoint to test would be Nashkel mines and similar, where you stay in the same worldmap location through several areas that are all likely in mastarea.2da.

Thanks for the info. I'll test this in the next round.